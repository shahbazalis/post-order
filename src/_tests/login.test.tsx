import React from "react";
import { render, RenderResult } from "@testing-library/react";
import Enzyme from "enzyme";
import LoginPage from "../views/login/login";

import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
const { createMemoryHistory } = require("history");
import { Router } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

Enzyme.configure({ adapter: new Adapter() });

describe("Login page tests", () => {
  const initialState = { output: 10 };
  const mockStore = configureStore();
  let history = createMemoryHistory();
  let wrapper: RenderResult;
  let store = mockStore(initialState);
  wrapper = render(
    <Provider store={store}>
      <Router location={history.location} navigator={history}>
        <LoginPage />
      </Router>
    </Provider>
  );
  it("snapshot login view", async () => {
    expect(wrapper).toMatchSnapshot();
  });
});
