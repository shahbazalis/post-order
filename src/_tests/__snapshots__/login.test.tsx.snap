// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page tests snapshot login view 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="div"
      >
        <div>
          <h1
            class="heading-1"
          >
            Login
          </h1>
        </div>
        <form>
          <input
            class="center-block"
            data-testid="input-name"
            name="name"
            placeholder="Name"
            type="text"
            value=""
          />
          <input
            class="center-block"
            data-testid="input-email"
            name="email"
            placeholder="Email"
            type="email"
            value=""
          />
          <div>
            <button
              class="center-block"
              data-testid="login-btn"
            >
              Go
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="div"
    >
      <div>
        <h1
          class="heading-1"
        >
          Login
        </h1>
      </div>
      <form>
        <input
          class="center-block"
          data-testid="input-name"
          name="name"
          placeholder="Name"
          type="text"
          value=""
        />
        <input
          class="center-block"
          data-testid="input-email"
          name="email"
          placeholder="Email"
          type="email"
          value=""
        />
        <div>
          <button
            class="center-block"
            data-testid="login-btn"
          >
            Go
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
